server/src/main/java/org/apache/iotdb/db/metadata/mtree/store/IMTreeStore.java

```diff
@@ -61,33 +61,33 @@ public interface IMTreeStore {
    * @param name name or alias
    * @return child node
    */
-  IMNode getChild(IMNode parent, String name) throws MetadataException;
+  N getChild(N parent, String name) throws MetadataException;
 
-  IMNodeIterator getChildrenIterator(IMNode parent) throws MetadataException;
+  IMNodeIterator<N> getChildrenIterator(N parent) throws MetadataException;
 
-  IMNodeIterator getTraverserIterator(
-      IMNode parent, Map<Integer, Template> templateMap, boolean skipPreDeletedSchema)
+  IMNodeIterator<N> getTraverserIterator(
+      N parent, Map<Integer, Template> templateMap, boolean skipPreDeletedSchema)
       throws MetadataException;
 
-  IMNode addChild(IMNode parent, String childName, IMNode child);
+  N addChild(N parent, String childName, N child);
 
-  void deleteChild(IMNode parent, String childName) throws MetadataException;
+  void deleteChild(N parent, String childName) throws MetadataException;
 
-  void updateMNode(IMNode node) throws MetadataException;
+  void updateMNode(N node) throws MetadataException;
 
-  IEntityMNode setToEntity(IMNode node) throws MetadataException;
+  IDeviceMNode<N> setToEntity(N node) throws MetadataException;
 
-  IMNode setToInternal(IEntityMNode entityMNode) throws MetadataException;
+  N setToInternal(IDeviceMNode<N> entityMNode) throws MetadataException;
 
-  void setAlias(IMeasurementMNode measurementMNode, String alias) throws MetadataException;
+  void setAlias(IMeasurementMNode<N> measurementMNode, String alias) throws MetadataException;
 
-  void pin(IMNode node) throws MetadataException;
+  void pin(N node) throws MetadataException;
 
-  void unPin(IMNode node);
+  void unPin(N node);
 
-  void unPinPath(IMNode node);
+  void unPinPath(N node);
```

Comment:
Maybe these interfaces should only be behavior of CachdMTreeStore. The common behavior are releaseNode and releasePath, which only used in Traverser.

Reply:
Yes.. but it's still be used in SchemaRegionSchemaFileImpl. I added these todo's about a month ago.. 

```
  // TODO: This interface should not be exposed to SchemaRegion
  public void pinMNode(ICacheMNode node) throws MetadataException {
    store.pin(node);
  }

  // TODO: This interface should not be exposed to SchemaRegion
  public void unPinMNode(ICacheMNode node) {
    store.unPin(node);
  }

  private void unPinPath(ICacheMNode node) {
    store.unPinPath(node);
  }

  // TODO: This interface should not be exposed to SchemaRegion
  public void updateMNode(ICacheMNode node) throws MetadataException {
    store.updateMNode(node);
  }
```

Reply:
At least the Traverser shall not sense these interface? It takes a big effort to refactor the implementations in SchemaRegionSchemaFileImpl.